### PR TITLE
TK-2514,TK-2515, TK-2519, TK-2520, TK-2537, TK-2521: preflight script and test suite bug fixes and improvements

### PR DESCRIPTION
**Pull request by msurudwad**
Fixed issues -
-  script stuck forever if no value provided for flags
-  fixed minimum helm and k8s version
-  removed support for tiller check
- removed feature gate check as it's not needed for k8s v1.17+
- added common label on resources created by Preflight Check
- cleanup stuck forever while cleanup
- Preflight script exiting and starting cleaning if one of checks fails in the middle
- removed volumesnapshot v1alpha1 related part as min. supported k8s v1.17.0 
- --snapshotclass flag made optional

### CODE REVIEW

File: tools/preflight/preflight.sh [hunk 1]
```diff
@@ -213,70 +216,40 @@ check_storage_snapshot_class() {
...
 
-  if [[ $(check_if_ocp) == "Y" ]]; then
-    features_enabled=$(kubectl get cm -n openshift-kube-apiserver -oyaml | grep feature)
+  # shellcheck disable=SC2143
+  if [[ $(kubectl get volumesnapshotclass | grep -E "(^|\s)${SNAPSHOT_CLASS}($|\s)") ]]; then
+    echo -e "${GREEN} ${CHECK} Volume snapshot class \"${SNAPSHOT_CLASS}\" found in cluster${NC}\n"
   else
-    features_enabled=$(kubectl get po -n kube-system -oyaml | grep feature)
+    echo -e "${RED} ${CROSS} Volume snapshot class \"${SNAPSHOT_CLASS}\" not found in cluster${NC}\n"
+    exit_status=1
```
Comment (madhurtrilio) [inline tools/preflight/preflight.sh:250]: Do we have a check if the user given SNAPSHOT_CLASS belongs to the STORAGE_CLASS?

File: tools/preflight/preflight.sh [hunk 2]
```diff
@@ -351,13 +324,20 @@ check_volume_snapshot() {
   local success_status=0
   local retries=30
   local sleep=5
-  set +o errexit
+
+  # shellcheck disable=SC2143
+  if [[ -z "${SNAPSHOT_CLASS}" ]]; then
+    echo -e "${RED} ${CROSS} Volume snapshot class not found with same driver of given storage class${NC}\n"
```
Comment (madhurtrilio) [inline tools/preflight/preflight.sh:330]: confused! Is the error statement correct?

Comment (msurudwad) [reply tools/preflight/preflight.sh:330]: Error statement is correct. Will update it, to make it understandable.

Basic meaning is, VSC which has driver same as SC's provisioner is not found in cluster.

File: tools/preflight/preflight.sh (hunk 1 above)
Comment (msurudwad) [reply tools/preflight/preflight.sh:250]: no, will make changes for it.

Comment (msurudwad) [reply tools/preflight/preflight.sh:258]: made changes.. please review

File: tools/preflight/preflight.sh (hunk 2 above)
Comment (msurudwad) [reply tools/preflight/preflight.sh]: made changes.. please review

**Review by madhurtrilio — APPROVED** [review]

**Pull request merged**